panel_app/panel_UI/panel_helpers.py

```diff
@@ -10,27 +10,109 @@
 from .config import *
 
 
+def _point_in_mask(nc_url, varname, point, latvar="lat", lonvar="lon", time_index=0):
+    """
+    Check if (lat, lon) is within [lat, lon] bounds of nc_url and not masked/missing
+    at the nearest grid cell for `varname`.
+    """
+    with Dataset(nc_url) as ds:
+        lat = ds.variables[latvar][:]
+        lon = ds.variables[lonvar][:]
+        plat, plon = float(point[0]), float(point[1])
+
+        if plat < lat[0] or plat > lat[-1] or plon < lon[0] or plon > lon[-1]:
+            return False
+
+        lat_index = int(np.argmin(np.abs(lat - plat)))
+        lon_index = int(np.argmin(np.abs(lon - plon)))
+
+        var = ds.variables[varname]
+        cell = (
+            var[time_index, lat_index, lon_index]
+            if getattr(var, "ndim", 2) == 3
+            else var[lat_index, lon_index]
+        )
+
+        # masked array?
+        if np.ma.isMaskedArray(cell) and np.ma.getmask(cell):
+            return False
+
+        # plain scalar: NaN or fill/missing?
+        try:
+            val = float(cell)
+        except Exception:
+            return True
+
+        if np.isnan(val):
+            return False
+
+        for attr in ("_FillValue", "missing_value"):
+            if hasattr(var, attr):
+                mv = getattr(var, attr)
+                mv_list = (
+                    list(mv)
+                    if np.iterable(mv) and not isinstance(mv, (str, bytes))
+                    else [mv]
+                )
+                for mvv in mv_list:
+                    try:
+                        if mvv is not None and float(val) == float(mvv):
+                            return False
+                    except Exception:
+                        pass
+        return True
+
+
 def in_bc(point):
-    """Check if a given point is within
-    the BC PRISM grid."""
-    bc = f"{THREDDS_BASE}/storage/data/climate/PRISM/dataportal/pr_monClim_PRISM_historical_run1_198101-201012.nc"
-    bc_data = Dataset(bc)
-    bc_lat = bc_data.variables["lat"][:]
-    bc_lon = bc_data.variables["lon"][:]
-    # Check if center point is within lat/lon grid
-    if (
-        (point[0] < bc_lat[0])
-        or (point[0] > bc_lat[-1])
-        or (point[1] < bc_lon[0])
-        or (point[1] > bc_lon[-1])
-    ):
-        return False
-    # Check if center point is closest to a masked data value
-    else:
-        lat_index = np.argmin(np.abs(bc_lat - point[0]))
-        lon_index = np.argmin(np.abs(bc_lon - point[1]))
-        pr = bc_data.variables["pr"][0, lat_index, lon_index]
-        if pr.mask:
+    url = f"{THREDDS_BASE}/storage/data/climate/PRISM/dataportal/pr_monClim_PRISM_historical_run1_198101-201012.nc"
```

Comment:
I wonder if these file locations could be cleaner or more configurable, they're being used in very generic functions